### PR TITLE
feat/page-with-create-datatable-tmplt-421

### PR DESCRIPTION
**Pull request by sventruschel**
Added datatable with create TMPLT-421

### CODE REVIEW

File: src/prefabs/pageWithCreateDataTable.js [hunk 1]
```diff
@@ -0,0 +1,12336 @@
...
+                'SIGNED_PDF',
+                'SUM',
+                'BOOLEAN_EXPRESSION',
+                'DATE_EXPRESSION',
+                'DATE_TIME_EXPRESSION',
+                'DECIMAL_EXPRESSION',
+                'INTEGER_EXPRESSION',
+                'MINUTES_EXPRESSION',
+                'PRICE_EXPRESSION',
+
```
Comment (jeroenreijs) [inline src/prefabs/pageWithCreateDataTable.js:6556]: It's better to define this list just once and use it at both PropertiesSelectors.

Comment (jeroenreijs) [inline src/prefabs/pageWithCreateDataTable.js:9324]: Your new `autoOK` option needs to be added here as well

Comment (jeroenreijs) [inline src/prefabs/pageWithCreateDataTable.js:9652]: Your new `autoOK` option needs to be added here as well

Comment (jeroenreijs) [inline src/prefabs/pageWithCreateDataTable.js:10016]: Your new `autoOK` option needs to be added here as well

**Review by jeroenreijs — CHANGES_REQUESTED** [review]

File: src/prefabs/pageWithCreateDataTable.js [hunk 2]
```diff
@@ -0,0 +1,12336 @@
...
+      createFormUseDataProperties,
+      setCreateFormUseDataProperties,
+    ] = React.useState(true);
+
+    const { data } = useModelQuery({
+      variables: { id: modelId },
+      skip: !modelId,
+    });
+
+    const reduceStructure = (refValue, structure) =>
```
Comment (jeroenreijs) [inline src/prefabs/pageWithCreateDataTable.js:44]: This is just a suggestion. it's not wrong like it is right now, see the other PR https://github.com/bettyblocks/material-ui-component-set/pull/1048. But it looks like a lookup for a certain prefab in the structure, though the function name is rather global. Maybe you can think of something more descriptive.

File: src/prefabs/pageWithCreateDataTable.js (hunk 1 above)
Comment (sventruschel) [reply src/prefabs/pageWithCreateDataTable.js:6562]: Defined it once and used it in both properties selectors

Comment (sventruschel) [reply src/prefabs/pageWithCreateDataTable.js:9330]: Added autoOK option

Comment (sventruschel) [reply src/prefabs/pageWithCreateDataTable.js:10022]: Added autoOK option

File: src/prefabs/pageWithCreateDataTable.js [hunk 3]
```diff
@@ -0,0 +1,12156 @@
...
+                />
+              </Field>
+            )}
+          </Field>
+        </Content>
+        <Footer
+          onSave={() => {
+            if (
+              !modelId ||
+              (createFormProperties.length < 1 &&
```
Comment (nulian) [inline src/prefabs/pageWithCreateDataTable.js:6405]: Why is length < 1 being done here. Feels like a weird way to check if array is empty.

The above `!createFormProperties.length` seems like a better way to check it.

Same for the cases below this.

**Review by nulian — CHANGES_REQUESTED** [review]

File: src/prefabs/pageWithCreateDataTable.js (hunk 3 above)
Comment (prorobbert) [reply src/prefabs/pageWithCreateDataTable.js]: That's a good feedback point. I'll try to keep using the shorter method from now on.

**Pull request closed without merging**